### PR TITLE
refactor: add ColumnDefinition type

### LINKED ISSUE #1193: AADev, I don't see any reference of viewFields in Table (should be named columnDefinitions)
- FieldMetadata + FieldDefinitions (editable-fields)
- Table uses columnDefinitions + FieldMetadata from editable-fields
- CompanyCardBoard uses FieldDefinitions + FieldMetadata

-----

ViewFields uses ViewField + FieldMetadata

-----
ex: Companies.tsx (page) will load ViewFields using views module, will do the mapping between ViewField and ColumnDefinition and will push to a columnDefnition state.

Table will only read into columnDefinitionState

### PR DESCRIPTION
**Pull request by thaisguigon**
Closes #1193

### CODE REVIEW

File: front/src/modules/companies/constants/companiesFieldDefinitions.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  IconMap,
+  IconMoneybag,
+  IconTarget,
+  IconUserCircle,
+  IconUsers,
+} from '@/ui/icon/index';
+import { Entity } from '@/ui/input/relation-picker/types/EntityTypeForSelect';
+import type { ColumnDefinition } from '@/ui/table/types/ColumnDefinition';
+
+export const companiesFieldDefinitions: ColumnDefinition<ViewFieldMetadata>[] =
```
Comment (charlesBochet) [inline front/src/modules/companies/constants/companiesFieldDefinitions.tsx:27]: companiesAvailableColumnDefinitions?

I think we should also add "available" in the naming to make clear it's the full list and remove any confusion with the one coming from the views

File: front/src/modules/people/constants/peopleFieldDefinitions.tsx [hunk 2]
```diff
@@ -21,14 +20,15 @@ import {
   IconUser,
 } from '@/ui/icon/index';
 import { Entity } from '@/ui/input/relation-picker/types/EntityTypeForSelect';
+import type { ColumnDefinition } from '@/ui/table/types/ColumnDefinition';
 
-export const peopleViewFields: ViewFieldDefinition<ViewFieldMetadata>[] = [
+export const peopleFieldDefinitions: ColumnDefinition<ViewFieldMetadata>[] = [
```
Comment (charlesBochet) [inline front/src/modules/people/constants/peopleFieldDefinitions.tsx:25]: peopleAvailableColumnDefinitions

File: front/src/modules/pipeline/constants/pipelineViewFields.tsx [hunk 3]
```diff
@@ -17,10 +17,8 @@ import { Entity } from '@/ui/input/relation-picker/types/EntityTypeForSelect';
 export const pipelineViewFields: ViewFieldDefinition<ViewFieldMetadata>[] = [
```
Comment (charlesBochet) [inline front/src/modules/pipeline/constants/pipelineViewFields.tsx:17]: maybe not in this PR which is already big, but we need to create a follow up ticket:
this file should be named pipelineFieldsDefinitions.

File: front/src/modules/ui/table/components/EntityTableCell.tsx [hunk 4]
```diff
@@ -26,17 +26,17 @@ export function EntityTableCell({ cellIndex }: { cellIndex: number }) {
     setContextMenuOpenState(true);
   }
 
-  const viewField = useContext(ViewFieldContext);
+  const columnDefinition = useContext(ColumnContext);
```
Comment (charlesBochet) [inline front/src/modules/ui/table/components/EntityTableCell.tsx:29]: awesome!

File: front/src/modules/ui/table/editable-cell/components/GenericEditableCell.tsx [hunk 5]
```diff
@@ -29,34 +27,36 @@ import { GenericEditableTextCell } from '../type/components/GenericEditableTextC
 import { GenericEditableURLCell } from '../type/components/GenericEditableURLCell';
 
 type OwnProps = {
-  viewField: ViewFieldDefinition<ViewFieldMetadata>;
+  fieldDefinition: ColumnDefinition<ViewFieldMetadata>;
```
Comment (charlesBochet) [inline front/src/modules/ui/table/editable-cell/components/GenericEditableCell.tsx:30]: fieldDefinition => columnDefinition?

File: front/src/modules/ui/table/hooks/useUpdateEntityField.ts [hunk 6]
```diff
@@ -73,7 +73,7 @@ export function useUpdateEntityField() {
       : unknown,
   >(
     currentEntityId: string,
-    viewField: ViewFieldDefinition<MetadataType>,
+    field: ColumnDefinition<MetadataType>,
```
Comment (charlesBochet) [inline front/src/modules/ui/table/hooks/useUpdateEntityField.ts:76]: field => columnDefinition here too

File: front/src/modules/companies/constants/companiesFieldDefinitions.tsx (hunk 1 above)
Comment (thaisguigon) [reply front/src/modules/companies/constants/companiesFieldDefinitions.tsx]: Done

File: front/src/modules/people/constants/peopleFieldDefinitions.tsx (hunk 2 above)
Comment (thaisguigon) [reply front/src/modules/people/constants/peopleFieldDefinitions.tsx]: Done

File: front/src/modules/pipeline/constants/pipelineViewFields.tsx (hunk 3 above)
Comment (thaisguigon) [reply front/src/modules/pipeline/constants/pipelineViewFields.tsx]: I renamed it `pipelineAvailableFieldDefinitions`.

File: front/src/modules/ui/table/editable-cell/components/GenericEditableCell.tsx (hunk 5 above)
Comment (thaisguigon) [reply front/src/modules/ui/table/editable-cell/components/GenericEditableCell.tsx]: Done

File: front/src/modules/ui/table/hooks/useUpdateEntityField.ts (hunk 6 above)
Comment (thaisguigon) [reply front/src/modules/ui/table/hooks/useUpdateEntityField.ts]: Done

**Review by charlesBochet — APPROVED** [review]
LGTM

**Pull request merged**